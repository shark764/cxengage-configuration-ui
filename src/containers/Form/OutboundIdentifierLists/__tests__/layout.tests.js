import React from 'react';
import { shallow } from 'enzyme';
import OutboundIdentifierListsForm from '../layout';

describe('OutboundIdentifierListsForm', () => {
  it('renders regular form', () => {
    const component = shallow(
      <OutboundIdentifierListsForm
        handleSubmit={() => 'mockFunctionReturn'}
        isSaving={false}
        inherited={false}
        flowIds={[]}
      />
    );
    expect(component).toMatchSnapshot();
  });
  it('renders , isSaving is true , fields should be disabled', () => {
    const component = shallow(
      <OutboundIdentifierListsForm
        handleSubmit={() => 'mockFunctionReturn'}
        isSaving={true}
        inherited={false}
        flowIds={[]}
      />
    );
    expect(component).toMatchSnapshot();
  });
  it('renders , inherited is true , fields should be disabled', () => {
    const component = shallow(
      <OutboundIdentifierListsForm
        handleSubmit={() => 'mockFunctionReturn'}
        isSaving={false}
        inherited={true}
        flowIds={[]}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
