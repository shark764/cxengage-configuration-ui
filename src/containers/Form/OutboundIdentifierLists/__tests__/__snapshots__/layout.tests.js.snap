// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutboundIdentifierListsForm renders , inherited is true , fields should be disabled 1`] = `
<form
  onSubmit={[Function]}
>
  <InputField
    componentType="input"
    disabled={true}
    inputType="text"
    label="Name *"
    name="name"
  />
  <InputField
    componentType="textarea"
    disabled={true}
    inputType="text"
    label="Description"
    name="description"
  />
</form>
`;

exports[`OutboundIdentifierListsForm renders , isSaving is true , fields should be disabled 1`] = `
<form
  onSubmit={[Function]}
>
  <InputField
    componentType="input"
    disabled={true}
    inputType="text"
    label="Name *"
    name="name"
  />
  <InputField
    componentType="textarea"
    disabled={true}
    inputType="text"
    label="Description"
    name="description"
  />
</form>
`;

exports[`OutboundIdentifierListsForm renders regular form 1`] = `
<form
  onSubmit={[Function]}
>
  <InputField
    componentType="input"
    disabled={false}
    inputType="text"
    label="Name *"
    name="name"
  />
  <InputField
    componentType="textarea"
    disabled={false}
    inputType="text"
    label="Description"
    name="description"
  />
</form>
`;
